util/topgen/intermodule.py

```diff
@@ -778,8 +778,19 @@ def check_intermodule(topcfg: Dict, prefix: str) -> int:
             err, rsp_struct = check_intermodule_field(rsp_struct)
             error += err
 
-            total_width += rsp_struct["width"]
-            widths.append(rsp_struct["width"])
+            if isinstance(rsp_struct["width"], Parameter):
+                param = rsp_struct["width"]
+                if param.expose:
+                    # If it's a top-level exposed parameter, we need to find definition from there
+                    module = lib.get_module_by_name(topcfg, req_m)
+                    width = int(module['param_decl'].get(param.name, rsp_struct["width"].default))
+                else:
+                    width = int(req_struct["width"].default)
```

Comment:
Do we know that `req_struct["width"]` has type `Parameter`? If so, maybe add an assertion here?

Reply:
There is a check `if isinstance(req_struct["width"], Parameter)` above or am I missing something?

Reply:
Do you mean line 781? That's looking at `rsp_struct`.

Reply:
that is more of an issue. Thanks, that has been fixed.